### PR TITLE
add semantic versioning (semver.org) validator

### PR DESCRIPTION
**Pull request by thiagowfx**
https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

### CODE REVIEW

**Comment by thiagowfx** [comment]
If you deem this PR worthwhile, please accept the workflow so that we can verify whether the tests pass.

**Comment by cblakkan** [comment]
Looks like you've got a typo in the test

```

      def test_semver():
          """
          - https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
          - https://regex101.com/r/Ly7O1x/3/
          """
  >       v = val.Semver()
  E       AttributeError: module 'yamale.validators' has no attribute 'Semver'. Did you mean: 'SemVer'?
  
```

**Comment by thiagowfx** [comment]
Ooops! Fixed

File: README.md [hunk 1]
```diff
@@ -400,6 +400,12 @@ Validates MAC addresses.
 Examples:
 - `mac()`: Allows any valid MAC address
 
+### SemVer (Semantic Versioning) - `SemVer()`
+Validates [Semantic Versioning](https://semver.org/) strings.
+
+Examples:
+- `SemVer()`: Allows any valid SemVer string
```
Comment (cblakkan) [inline README.md:407]: I think this should be `semver()` instead of `SemVer()` right?

Comment (thiagowfx) [reply README.md]: Looking at the rest of the values, for consistency, yes. Updated.

Although, intuitively, I'd have expected the UpperCamelCase version to be the correct one, to mirror the class definition. I suppose it uses the value of `tag` then.

**Pull request merged**